File: DSA/JavaScript/StacksAndQueues/Stack.js
class Stack {
	constructor(size) {
		this.size = size;
		this.arr = new Array(this.size);
		this.top = -1;
	}

	getSize() {
		console.log("Stack Size : ", this.top + 1);
		return this.top + 1;
	}

	peek() {
		console.log("Top Element is : ", this.arr[this.top]);
		return this.arr[this.top];
	}

	isEmpty() {
		if (this.top == -1) {
			console.log("Stack is Empty !");
			return true;
		} else {
			console.log(" Stack is not Empty !");
			return false;
		}
	}

	empty() {
		this.top = -1;
		console.log("Emptied  Stack !");
	}

	push(value) {
		if (this.top >= this.size - 1) {
			console.log("Stack Overflow : Size Limit Exceeded !!");
			return;
		}
		this.top++;
		this.arr[this.top] = value;
	}

	pop() {
		if (this.isEmpty()) {
			return -1;
		}
		let deletedValue = this.arr[this.top];
		this.top--;
		return deletedValue;
	}

	printStack() {
		if (this.isEmpty()) {
			console.log("-> X");
			return;
		}
		console.log();
		console.log("Stack : ");
		process.stdout.write(" -> ");
		console.log(`${this.arr[this.top]}`);
		for (let i = this.top - 1; i >= 0; i--) {
			console.log(`    ${this.arr[i]}`);
		}
		console.log();
	}
}

function main() {
	const stack = new Stack(5);
	stack.isEmpty();

	stack.push(1);
	stack.push(2);
	stack.push(3);
	stack.push(4);
	stack.push(5);

	stack.peek();
	stack.getSize();
	stack.isEmpty();
	stack.printStack();

	stack.empty();
	stack.printStack();

	stack.push(6);
	stack.push(7);
	stack.push(8);
	stack.push(9);
	stack.push(10);
	stack.printStack();

	stack.pop();
	stack.printStack();
	stack.peek();
	stack.getSize();
	return 0;
}

main();

File: DSA/CPP/StacksAndQueues/Stack.cpp
#include <iostream>
using namespace std;

class Stack{
    public:
        int* arr;
        int size = 0;
        int top = -1;
       
        Stack(int size){
            this->size = size;
            arr = new int[size];
        }
       
       
        int getSize(){
            cout << "Stack Size : " << top + 1 << endl;
            return top + 1;
        }
       
        int peek(){
            cout << "Top Element is : " << arr[top] << endl;
            return arr[top];
        }
       
        bool isEmpty(){
            if(top == -1){
                cout << "Stack is Empty !" << endl;
                return true;
            } else {
                cout << " Stack is not Empty !" << endl;
                return false;
            }
        }
       
        void empty(){
            top = -1;
            cout << "Emptied  Stack !" << endl;
        }
       
        void push(int value){
            if(top >= size - 1) {
                cout << "Stack Overflow : Size Limit Exceeded !!" << endl;
                return;
            }
            top++;
            arr[top] = value;
        }
       
        int pop(){
            if(this->isEmpty()){
                return -1;
            }
            int deletedValue = arr[top];
            top--;
            return deletedValue;
        }
       
        void printStack(){
            if(this->isEmpty()){
                 cout << "-> X" << endl;
                 return;
            }
            cout << endl << "Stack : " << endl;
            cout <<  "-> " << arr[top] << endl;
            for(int i = top - 1; i >= 0; i--){
                cout << "   " << arr[i] << endl;
            }
            cout << endl;
        }
};

int main(){
    Stack* stack = new Stack(5);
    stack->isEmpty();
   
    stack->push(1);
    stack->push(2);
    stack->push(3);
    stack->push(4);
    stack->push(5);
   
    stack->peek();
    stack->getSize();
    stack->isEmpty();
    stack->printStack();
   
    stack->empty();
    stack->printStack();
   
    stack->push(6);
    stack->push(7);
    stack->push(8);
    stack->push(9);
    stack->push(10);
    stack->printStack();
   
    stack->pop();
    stack->printStack();
    stack->peek();
    stack->getSize();
    return 0;
}